docs/features/nullable-metadata.md

```diff
@@ -0,0 +1,189 @@
+- Reference type: the nullability (0, 1, or 2), followed by the representation of the type arguments in order including containing types
+- Nullable value type: the representation of the type argument only
+- Value type: the representation of the type arguments in order including containing types
+- Array: the nullability (0, 1, or 2), followed by the representation of the element type
+- Tuple: the representation of the tuple elements in order
+- Type parameter reference: the nullability (0, 1, or 2, with 0 for unconstrained type parameter)
+
+Note that certain type references are represented by an empty `byte[]`:
+- Non-generic value types, and
+- Nullable value types and tuples where all elements are represented by an empty `byte[]`.
+
+### Optimizations
+
+If the `byte[]` is empty, the `NullableAttribute` is omitted.
+
+If all values in the `byte[]` are the same, the `NullableAttribute` is constructed with that single `byte` value. (For instance, `NullableAttribute(1)` rather than `NullableAttribute(new byte[] { 1, 1 }))`.)
+
+### Type parameters
+Each type parameter definition may have an associated `NullableAttribute` with a single `byte`:
+
+1. `notnull` constraint: `NullableAttribute(1)`
+2. `class` constraint in `#nullable disable` context: `NullableAttribute(0)`
+3. `class` constraint in `#nullable enable` context: `NullableAttribute(1)`
+4. `class?` constraint: `NullableAttribute(2)`
+5. No `notnull`, `class`, `struct`, `unmanaged`, or type constraints in `#nullable disable` context: `NullableAttribute(0)`
+6. No `notnull`, `class`, `struct`, `unmanaged`, or type constraints in `#nullable enable` context
+(equivalent to an `object?` constraint): `NullableAttribute(2)`
+
+## NullableContextAttribute
+`NullableContextAttribute` can be used to indicate the nullability of type references that have no `NullableAttribute` annotations.
+
+```C#
+namespace System.Runtime.CompilerServices
+{
+    [System.AttributeUsage(
+        AttributeTargets.Module |
+        AttributeTargets.Class |
+        AttributeTargets.Delegate |
+        AttributeTargets.Field |
+        AttributeTargets.Interface |
+        AttributeTargets.Method |
+        AttributeTargets.Property |
+        AttributeTargets.Struct,
+        AllowMultiple = false,
+        Inherited = false)]
+    public sealed class NullableContextAttribute : Attribute
+    {
+        public readonly byte Flag;
+        public NullableContextAttribute(byte flag)
+        {
+            Flag = flag;
+        }
+    }
+}
+```
+
+The `NullableContextAttribute` type is for compiler use only - it is not permitted in source.
+The type declaration is synthesized by the compiler if not already included in the compilation.
+
+The `NullableContextAttribute` is optional - nullable annotations can be represented in metadata with full fidelity using `NullableAttribute` only.
+
+`NullableContextAttribute` is valid in metadata at the module level and at type and method declarations.
```

Comment:
It looks like Fields and properties are listed as valid targets, but they are not methods. Consider removing them from the list.